pkg/controller/kibana/controller_test.go

```diff
@@ -0,0 +1,389 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package kibana
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	commonv1 "github.com/elastic/cloud-on-k8s/pkg/apis/common/v1"
+	esv1 "github.com/elastic/cloud-on-k8s/pkg/apis/elasticsearch/v1"
+	kibanav1 "github.com/elastic/cloud-on-k8s/pkg/apis/kibana/v1"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/operator"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/watches"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
+)
+
+//nolint:thelper
+func TestReconcileKibana_Reconcile(t *testing.T) {
+	sampleElasticsearch := esv1.Elasticsearch{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-es",
+			Namespace: "test",
+		},
+		Spec: esv1.ElasticsearchSpec{
+			Version: "7.17.0",
+		},
+	}
+	sampleKibana := kibanav1.Kibana{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:       "test-kibana",
+			Namespace:  "test",
+			Generation: 2,
+		},
+		Spec: kibanav1.KibanaSpec{
+			Version: "7.17.0",
+			Count:   1,
+		},
+		Status: kibanav1.KibanaStatus{
+			ObservedGeneration: 1,
+		},
+	}
+	defaultRequest := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      "test-kibana",
+			Namespace: "test",
+		},
+	}
+	type fields struct {
+		Client         k8s.Client
+		recorder       record.EventRecorder
+		dynamicWatches watches.DynamicWatches
+		params         operator.Parameters
+	}
+	tests := []struct {
+		name     string
+		fields   fields
+		request  reconcile.Request
+		want     reconcile.Result
+		wantErr  bool
+		validate func(*testing.T, fields)
+	}{
+		{
+			name: "unmanaged kibana instance does increment observedGeneration",
+			fields: fields{
+				Client: k8s.NewFakeClient(
+					&sampleElasticsearch,
+					withAnnotations(&sampleKibana, map[string]string{common.ManagedAnnotation: "false"}),
+				),
+				recorder:       record.NewFakeRecorder(100),
+				dynamicWatches: watches.NewDynamicWatches(),
+				params:         operator.Parameters{},
```

Comment:
These three parameters are always the same you might as well move them into the test loop instead of listing them for every test.